src/libstd/dynamic_lib.rs

```diff
@@ -51,8 +52,8 @@ impl DynamicLibrary {
 
     /// Lazily open a dynamic library. When passed None it gives a
     /// handle to the calling process
-    pub fn open(filename: Option<&Path>) -> Result<DynamicLibrary, String> {
-        let maybe_library = dl::open(filename.map(|path| path.as_vec()));
+    pub fn open<P: AsPath + ?Sized>(filename: Option<&P>) -> Result<DynamicLibrary, String> {
```

Comment:
Unfortunately using a type parameter I believe will cause `DynamicLibrary::open(None)` to fail as there's no way to determine the type of `P`. Perhaps this could stay as `Path` for now?